### PR TITLE
Merging of OAuth + Connections

### PR DESCRIPTION
**Pull request by partydragen**
Merge OAuth into connections instead of 2 seperate linking systems making alot of users confused and to not need to link two times on same provider

Aswell to solve current problems releated with connections using oauth for linking aswell
Also its confusing to setup integration that requires oauth for then you need to setup oauth + integration instead of just doing it all in integration page

Progress:
- [x] Remove OAuth pages
- [x] Make users login/register using the connections data
- [x] Add google integration
- [x] Move OAuth settings to integration setting page instead of squeeze it all into same page and single row
- [x] Add option to allow user to login/register with the oauth (Currently if you add linking with it it also requires login/register)

### CODE REVIEW

File: core/classes/Misc/NamelessOAuth.php [hunk 1]
```diff
@@ -46,19 +46,18 @@ public function getProviders(): array
...
     /**
-     * Determine if OAuth is available if at least one provider is setup.
+     * Get or create an instance of a specific provider.
      *
-     * @return bool If any provider is setup
+     * @param string $provider The provider name
+     * @return null|array An array of registered OAuth provider.
      */
-    public function isAvailable(): bool
+    public function getProvider(string $provider): ?array
```
Comment (samerton) [inline core/classes/Misc/NamelessOAuth.php:54]: @partydragen can you confirm that this is correct please - this was the one merge conflict I had to resolve for this PR

File: custom/panel_templates/Default/core/integrations_edit.tpl [hunk 2]
```diff
@@ -88,6 +88,70 @@
...
+                                                <label for="enable" id="enable"
+                                                       class="custom-control-label">
+                                                    {$REGISTER_LOGIN_WITH_OAUTH}
+                                                </label>
+                                            </div>
+
+                                            <div class="form-group">
+                                                <label for="client-id">{$CLIENT_ID}</label>
+                                                <input type="text" name="client-id" class="form-control"
+                                                       id="client-id" placeholder="Client ID"
```
Comment (samerton) [inline custom/panel_templates/Default/core/integrations_edit.tpl]: Some hardcoded terms in the placeholders here - `Client ID`, `Client Secret`, and then there's no placeholder at all for the client URL below

File: core/classes/Misc/NamelessOAuth.php (hunk 1 above)
Comment (partydragen) [reply core/classes/Misc/NamelessOAuth.php:54]: I have no idea however it looks to be working

File: custom/panel_templates/Default/core/integrations_edit.tpl (hunk 2 above)
Comment (partydragen) [reply custom/panel_templates/Default/core/integrations_edit.tpl]: Fixed, Also Client URI is never empty as its set by NamelessMC so no Placeholder needed

**Review by samerton — APPROVED** [review]

**Pull request merged**